app/src/main/java/com/asfoundation/wallet/nfts/ui/nftdetails/NFTDetailsFragment.kt

```diff
@@ -0,0 +1,103 @@
+package com.asfoundation.wallet.nfts.ui.nftdetails
+
+import android.graphics.drawable.Drawable
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.ImageView
+import androidx.fragment.app.viewModels
+import androidx.lifecycle.lifecycleScope
+import by.kirich1409.viewbindingdelegate.viewBinding
+import com.asf.wallet.R
+import com.asf.wallet.databinding.FragmentNftBinding
+import com.asfoundation.wallet.base.SingleStateFragment
+import com.asfoundation.wallet.viewmodel.BasePageViewFragment
+import com.bumptech.glide.Glide
+import com.bumptech.glide.load.DataSource
+import com.bumptech.glide.load.engine.GlideException
+import com.bumptech.glide.request.RequestListener
+import javax.inject.Inject
+
+class NFTDetailsFragment : BasePageViewFragment() ,
+  SingleStateFragment<NFTDetailsState, NFTDetailsSideEffect> {
+
+  @Inject
+  lateinit var viewModelFactory: NFTDetailsViewModelFactory
+
+  @Inject
+  lateinit var navigator: NFTDetailsNavigator
+
+  private val viewModel: NFTDetailsViewModel by viewModels { viewModelFactory }
+  private val views by viewBinding(FragmentNftBinding::bind)
+
+  override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
+                            savedInstanceState: Bundle?): View? {
+    postponeEnterTransition()
+    return inflater.inflate(R.layout.fragment_nft, container, false)
+  }
+
+  override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+    super.onViewCreated(view, savedInstanceState)
+    views.nftTitle.text = viewModel.state.data.name
+    views.nftSubtitle.text = viewModel.state.data.description
+    views.nftImage.load(viewModel.state.data.imageURL) {
+      startPostponedEnterTransition()
+    }
```

Comment:
This shouldn't be used like this here, these assignments should be done in onStateChanged()